packages/backend-dynamic-feature-service/src/scanner/plugin-scanner.ts

```diff
@@ -313,3 +321,70 @@ export class PluginScanner {
     this.untrackChanges();
   }
 }
+
+/**
+ * @public
+ */
+export interface DynamicPluginsSchemaDiscoveryOptions {
+  /**
+   * Function that returns the plugin-relative path to the Json schema file for a given platform.
+   * Default behavior is to look for the `configSchema.json` file in the package `dist` sub-directory.
+   *
+   * @param platform - The platform of the plugin.
+   * @returns the plugin-relative path to the Json schema file.
+   */
+  schemaLocator?: (platform: PackagePlatform) => string;
```

Comment:
`PackagePlatform` is just `node`, `web` or `common`. I think this could be something like `schemaLocationByPlatform: { [platform in PackagePlatform]: path }` instead?

Reply:
> I think this could be something like schemaLocationByPlatform: { [platform in PackagePlatform]: path } instead

Wouldn't doing this force us to duplicate the value when it is the same value for both cases ?

As an example the `schemaLocator` argument default value in `gatherDynamicPlginsSchemas()` would change from:

```ts
  schemaLocator: (platform: PackagePlatform) => string = () =>
    path.join('dist', 'configSchema.json'),
```
to
```ts
  schemaLocationByPlatform: { [platform in PackagePlatform]: string } = {
    'web': path.join('dist', 'configSchema.json'),
    'node': path.join('dist', 'configSchema.json'),
  }
```
I can do the change if you prefer, but I'm not convinced it is better.
I'd rather give even more freedom to the caller and pass the whole `ScannedPluginPackage` as an argument.

Reply:
Yeah it was more a comment on the callback not being very powerful tbh. I think it make sense to provide more information instead, and keep the callback.

Reply:
Let me change it.

Reply:
Done in commit a9e95d6ad1770745c6493de87965d97bf1090ccd